docs/Measuring_Resonances.md

```diff
@@ -65,45 +41,116 @@ the system that may damage the electronics.
 
 ## Software installation
 
-Note that resonance measurements and shaper auto-calibration require additional
-software dependencies not installed by default. First, you will have to run on
-your Raspberry Pi the following command:
+Because setting up the software for measuring resonances takes significant
+time and resources, it is not installed by default. Run the following
+script to setup `numpy` and `matplotlib` to get started:
+
 ```
-~/klippy-env/bin/pip install -v numpy
+~/klipper/scripts/setup-shaper-calibrate.sh
 ```
-to install `numpy` package. Note that, depending on the performance of the
+
+Note that, depending on the performance of the
 CPU, it may take *a lot* of time, up to 10-20 minutes. Be patient and wait
 for the completion of the installation. On some occasions, if the board has
-too little RAM, the installation may fail and you will need to enable swap.
+too little RAM, the installation may fail, and you will need to
+[enable swap](https://nebl.io/neblio-university/enabling-increasing-raspberry-pi-swap/).
 
-Next, run the following commands to install the additional dependencies:
-```
-sudo apt update
-sudo apt install python-numpy python-matplotlib
-```
+## Wiring Option 1: Direct Raspberry Pi Connection
+
+The recommended way to connect the sensor to the printer is to connect it directly to
+the SPI bus on the Raspberry Pi.
+
+You need to connect ADXL345 to your Raspberry Pi via SPI. Note that the I2C
+connection, which is suggested by ADXL345 documentation, has too low throughput
+and **will not work**. The recommended connection scheme:
+
+| ADXL345 pin  | RPi pin | RPi pin name |
+| :--:         | :--:    | :--: |
+| 3V3 (or VCC) | 01      | 3.3v DC power |
+| GND          | 06      | Ground |
+| CS           | 24      | GPIO08 (SPI0_CE0_N) |
+| SDO          | 21      | GPIO09 (SPI0_MISO) |
+| SDA          | 19      | GPIO10 (SPI0_MOSI) |
+| SCL          | 23      | GPIO11 (SPI0_SCLK) |
+
+Fritzing wiring diagrams for some of the ADXL345 boards:
+
+![ADXL345-Rpi](img/adxl345-fritzing.png)
 
-Afterwards, check and follow the instructions in the
-[RPi Microcontroller document](RPi_microcontroller.md) to setup the
-"linux mcu" on the Raspberry Pi.
+Double-check your wiring before powering up the Raspberry Pi to prevent
+damaging it or the accelerometer.
 
-Make sure the Linux SPI driver is enabled by running `sudo
-raspi-config` and enabling SPI under the "Interfacing options" menu.
+Check and follow the
+instructions in the [RPi Microcontroller document](RPi_microcontroller.md) to setup the
+"linux mcu" on the Raspberry Pi. Make sure the Linux SPI driver is enabled by running
+`sudo raspi-config` and enabling SPI under the "Interfacing options" menu.
 
 Add the following to the printer.cfg file:
+
 ```
 [mcu rpi]
 serial: /tmp/klipper_host_mcu
 
 [adxl345]
-cs_pin: rpi:None
+cs_pin: rpi: None
```

Comment:
I'm not sure about the space here. I think most of examples don't have a space between the mcu name and the pin name. So let's not introduce it here.